File: Python/SearchInsertPosition.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        start = 0
        end = len(nums) - 1
        def find(target, start, end):
            if start > end:
                return start
            middle = int((start + end)/2)
            
            if target == nums[middle]:
                return middle
            elif nums[middle] > target:
                return find(target, start, middle-1)
            elif nums[middle] < target:
                return find(target, middle+1, end)
         
        return find(target, start, end)

File: C/Search Insert Position.c
int searchInsert(int* nums, int numsSize, int target) {
    int low = 0;
    int high = numsSize-1 ;
    int mid = 0;

    while(nums[mid] != target){
        mid = low + (high - low)/2;
        if(nums[mid] == target){
            return mid;
        }
        else if(nums[mid] > target){
            high = mid - 1;
        }
        else{
            low = mid + 1;
        }

        if(low > high){break;}

    }
    return low;
}